extensions/sql-database-projects/src/tools/netcoreTool.ts

```diff
@@ -96,13 +125,64 @@ export class NetCoreTool {
 		return undefined;
 	}
 
+	/**
+	 * This function checks if the installed dotnet version is 3.1 or greater.
+	 * Versions lower than 3.1 aren't supported for building projects.
+	 * Returns: True if installed dotnet version is supported, false otherwise.
+	 * 			Undefined if dotnet isn't installed.
+	 */
+	private async isNetCoreVersionSupported(): Promise<boolean | undefined> {
+		try {
+			const spawn = child_process.spawn;
+			let child: child_process.ChildProcessWithoutNullStreams;
+			let isSupported: boolean | undefined = undefined;
+
+			child = spawn('dotnet --version', [], {
+				shell: true
+			});
+
+			child.stdout.on('data', (data) => {
```

Comment:
I don't think we should be assuming that when we get a data event it's going to be the entire string we need. 

https://nodejs.org/api/stream.html#stream_class_stream_readable

Generally you should be batching up the data and then when you get the stream exit event take that buffer and turn it into a string then. 

See https://github.com/microsoft/azuredatastudio/blob/main/extensions/azdata/src/common/childProcess.ts#L51 for an example